src/test/scala/beam/integration/CAVSpec.scala

```diff
@@ -94,4 +96,31 @@ class CAVSpec extends AnyFlatSpec with Matchers with BeamHelper {
     assert(cavTrips >= cavVehicles, "Not enough CAV trips (by mode choice) seen.")
   }
 
+  val retries = 5
+
+  override def withFixture(test: NoArgTest) = {
+    if (isRetryable(test))
+      withFixture(test, retries)
+    else
+      super.withFixture(test)
+  }
+
+  def withFixture(test: NoArgTest, count: Int): Outcome = {
+    val outcome = super.withFixture(test)
+    println(outcome.toString)
+    outcome match {
+      case Failed(_) =>
+        if (count == 1) {
+          super.withFixture(test)
+        } else
+          withFixture(test, count - 1)
+      case Canceled(_) =>
+        if (count > 0) {
+          super.withFixture(test)
+        } else
+          withFixture(test, count - 1)
+      case other => other
+    }
+  }
```

Comment:
I would rewrite this like that:

  def withFixture(test: NoArgTest, count: Int): Outcome = {
    val outcome = super.withFixture(test)
    println(outcome.toString)
    outcome match {
      case Failed(_) if count==1 => super.withFixture(test)
      case Failed(_) =>	             withFixture(test, count - 1)
      case Canceled(_) if count ==1 => super.withFixture(test)
       case Canceled(_) =>                    withFixture(test, count - 1)
      case other => other
    }
  }